src/java.compiler/share/classes/javax/tools/JavaCompiler.java

```diff
@@ -181,32 +186,7 @@
  *     example, here is how to define a file object which represent
  *     source code stored in a string:
  *
- *     <pre>
- *       /**
- *        * A file object used to represent source coming from a string.
- *        {@code *}/
- *       public class JavaSourceFromString extends SimpleJavaFileObject {
- *           /**
- *            * The source code of this "file".
- *            {@code *}/
- *           final String code;
- *
- *           /**
- *            * Constructs a new JavaSourceFromString.
- *            * {@code @}param name the name of the compilation unit represented by this file object
- *            * {@code @}param code the source code for the compilation unit represented by this file object
- *            {@code *}/
- *           JavaSourceFromString(String name, String code) {
- *               super({@linkplain java.net.URI#create URI.create}("string:///" + name.replace('.','/') + Kind.SOURCE.extension),
- *                     Kind.SOURCE);
- *               this.code = code;
- *           }
- *
- *           {@code @}Override
- *           public CharSequence getCharContent(boolean ignoreEncodingErrors) {
- *               return code;
- *           }
- *       }</pre>
+ *     {@snippet id=fileObject class=JavaSourceFromString }
```

Comment:
JEP 413 uses the "file" attribute for external snippets, so using "class" here is new (to me anyway). Maybe the JEP should include an example that uses it.

Is this the first use of an external snippet in the src tree? I suspect files in the snippet-files directory will need a copyright header (downstream buildes will report issues on this). It's not clear to me how that works if @replace region replacement="" isn't the first line of the external file.

Reply:
Added a legal header in the external snippet file and tested with a full `:tier1` build, including `validate-source`.